src/app/core/entity-components/entity-list/entity-list.component.ts

```diff
@@ -206,6 +212,46 @@ export class EntityListComponent<T extends Entity>
     this.paginatorPageSize = event.pageSize;
     this.paginatorPageIndex = event.pageIndex;
     this.updateUserPaginationSettings();
+    this.showAllToggle =
+      this.paginatorPageSize >= this.entityDataSource.data.length
+        ? true
+        : false;
+  }
+
+  getPaginatorPageSizeOptions(): number[] {
+    const ar = [3, 10, 20, 50].filter((n) => {
+      return n < this.entityDataSource.data.length;
+    });
+    ar.push(this.entityDataSource.data.length);
+    return ar;
+  }
+
+  getPaginatorPageSize(): number {
+    if (
+      this.entityDataSource.data.length &&
+      this.paginatorPageSize >= this.entityDataSource.data.length
+    ) {
+      this.paginatorPageSize = this.entityDataSource.data.length;
+    }
+    return this.paginatorPageSize;
+  }
+
+  clickShowAllToggle() {
+    if (!this.showAllToggle) {
+      this.paginatorPageSizeBeforeToggle = this.paginatorPageSize;
+      this.paginatorPageSize = this.entityDataSource.data.length;
+    } else {
+      if (
+        this.paginatorPageSizeBeforeToggle <= this.entityDataSource.data.length
+      ) {
+        this.paginatorPageSize = this.paginatorPageSizeBeforeToggle;
+      } else {
+        const po = this.getPaginatorPageSizeOptions();
```

Comment:
I think you could also access the pagination session that is saved in the user entity.

Reply:
How do you mean that? Do you think that I don't need the paginatorPageSizeBeforeToggle? But imho the pagination session that is saved in the user entity only holds the current value, not the historic one.

Reply:
Currently, the page size is only saved on the user entity if the dropdown is used but not with the toggle. I think the page size should be saved in both cases.

Reply:
I've added pdateUserPaginationSettings() at the end of clickShowAllToggle.